src/puppeteer.test.ts

```diff
@@ -66,4 +67,14 @@ describe('Puppeteer Service', () => {
     const actualPdfContent = await pdf(result);
     expect(actualPdfContent.text).toEqual(expectedPdfContent.text);
   });
+
+  it('should not visit local urls', async () => {
+    const url = `file://${path.resolve(TEST_HTML_URL)}`;
+
+    await expect(async () => {
+      process.env.NODE_ENV = Environment.PRODUCTION;
```

Comment:
Imo sinon could be better solution ;)

Reply:
As far as i see it's very specific problem, because normally this function should not throw in test environment, because we want to use local files.

But otherwise it should fail. So if we want to test it failing, we need to simulate other environment.

I would propose another solution for that. Instead of relying on environment (test/prod), I would create env configuration flag, that will toggle url validation. 

That would make more sense imho, and be more reusable. For example it may be desired for someone to allow local URLs, although by default this should be disabled.

@mjavor what do you think and maybe you can make that change?

Reply:
Sure, I'll change it.